test-tools/aot-analyzer/src/main.cc

```diff
@@ -0,0 +1,656 @@
+
+    for (uint32_t index = 0; index < memory_count; index++) {
+        AOTMemory memory = memories[index];
+        printf("  -[%u] memory_flags:%5u    bytes_per_page:%5u    "
+               "init_page_count:%5u    max_page_count:%5u \n",
+               index, memory.memory_flags, memory.num_bytes_per_page,
+               memory.mem_init_page_count, memory.mem_max_page_count);
+    }
+    printf("\n\n");
+
+    // Globals
+    const uint32_t global_count =
+        ((AOTModule *)(aot->GetModule()))->global_count;
+    AOTGlobal *globals = ((AOTModule *)(aot->GetModule()))->globals;
+    printf("Globals[%u] \n", global_count);
+
+    for (uint32_t index = 0; index < global_count; index++) {
+        AOTGlobal global = globals[index];
+        printf("  -[%u] ", index);
+        printf("type: ");
+        dump_value_type(global.type);
+        printf("    is_mutable: %d    size: %u    data_offset: %u \n",
+               global.is_mutable, global.size, global.data_offset);
+    }
+    printf("\n\n");
+
+    // Exports
+    const uint32_t export_count =
+        ((AOTModule *)(aot->GetModule()))->export_count;
+    AOTExport *exports = ((AOTModule *)(aot->GetModule()))->exports;
+    printf("Exports[%u] \n", export_count);
+
+    for (uint32_t index = 0; index < export_count; index++) {
+        AOTExport expt = exports[index];
+        printf("  -[%u] kind:%5d    index:%5u    name: %s \n", index, expt.kind,
+               expt.index, expt.name);
+    }
+    printf("\n\n");
+
+    // Code
+    const uint32_t aot_code_size = (aot->GetMemConsumption()).aot_code_size;
+    const uint32_t literal_size =
+        ((AOTModule *)(aot->GetModule()))->literal_size;
+    const uint32_t data_section_count =
+        ((AOTModule *)(aot->GetModule()))->data_section_count;
+
+    printf("Codes[%u] \n", aot_code_size);
+    printf("  -code \n");
+    printf("    -code_size: %u Bytes \n", code_size);
+    printf("\n");
+    printf("  -literal \n");
+    printf("    -literal_size: %u Bytes \n", literal_size);
+    printf("\n");
+    printf("  -data section \n");
+    for (uint32_t index = 0; index < data_section_count; index++) {
+        AOTObjectDataSection *obj_data =
+            ((AOTModule *)(aot->GetModule()))->data_sections + index;
+        printf("    -[%u] code_size:%5u Bytes    name: %s \n", index,
+               obj_data->size, obj_data->name);
+    }
+    printf("\n\n");
+}
+
+void
+DumpCompare(AoTFile *aot, WasmFile *wasm)
+{
+    const AOTTargetInfo target_info = aot->GetTargetInfo();
+    AOTModule *aot_module = (AOTModule *)(aot->GetModule());
+    WASMModuleMemConsumption aot_mem_conspn = aot->GetMemConsumption();
+
+    WASMModule *wasm_module = (WASMModule *)(wasm->GetModule());
+
+    const uint32_t aot_func_count = aot_module->func_count;
+    const uint32_t aot_code_size = aot_module->code_size;
+    struct func_info *sorted_func_ptrs = NULL;
+    sorted_func_ptrs = sort_func_ptrs(((AOTModule *)(aot->GetModule())));
```

Comment:
Had better check whether sorted_func_ptrs is NULL?

Reply:
done